demo/admin/src/products/generated/ProductsGrid.tsx

```diff
@@ -0,0 +1,229 @@
+// This file has been generated by comet admin-generator.
+// You may choose to use this file as scaffold by moving this file out of generated folder and removing this comment.
+import { useApolloClient, useQuery } from "@apollo/client";
+import {
+    CrudContextMenu,
+    GridFilterButton,
+    muiGridFilterToGql,
+    muiGridSortToGql,
+    StackLink,
+    Toolbar,
+    ToolbarAutomaticTitleItem,
+    ToolbarFillSpace,
+    ToolbarItem,
+    useBufferedRowCount,
+    useDataGridRemote,
+    usePersistentColumnState,
+} from "@comet/admin";
+import { Add as AddIcon, Edit } from "@comet/admin-icons";
+import { BlockPreviewContent } from "@comet/blocks-admin";
+import { DamImageBlock } from "@comet/cms-admin";
+import { Box, Button, IconButton } from "@mui/material";
+import { DataGridPro, GridColDef, GridToolbarQuickFilter } from "@mui/x-data-grid-pro";
+import gql from "graphql-tag";
+import * as React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import {
+    GQLCreateProductMutation,
+    GQLCreateProductMutationVariables,
+    GQLDeleteProductMutation,
+    GQLDeleteProductMutationVariables,
+    GQLProductsGridQuery,
+    GQLProductsGridQueryVariables,
+    GQLProductsListFragment,
+} from "./ProductsGrid.generated";
+
+const productsFragment = gql`
+    fragment ProductsList on Product {
+        id
+        updatedAt
+        title
+        visible
+        slug
+        description
+        type
+        price
+        inStock
+        image
+        createdAt
+    }
+`;
+
+const productsQuery = gql`
+    query ProductsGrid($offset: Int, $limit: Int, $sort: [ProductSort!], $search: String, $filter: ProductFilter) {
+        products(offset: $offset, limit: $limit, sort: $sort, search: $search, filter: $filter) {
+            nodes {
+                ...ProductsList
+            }
+            totalCount
+        }
+    }
+    ${productsFragment}
+`;
+
+const deleteProductMutation = gql`
+    mutation DeleteProduct($id: ID!) {
+        deleteProduct(id: $id)
+    }
+`;
+
+const createProductMutation = gql`
+    mutation CreateProduct($input: ProductInput!) {
+        createProduct(input: $input) {
+            id
+        }
+    }
+`;
+
+function ProductsGridToolbar() {
+    return (
+        <Toolbar>
+            <ToolbarAutomaticTitleItem />
+            <ToolbarItem>
+                <GridToolbarQuickFilter />
+            </ToolbarItem>
+            <ToolbarFillSpace />
+            <ToolbarItem>
+                <GridFilterButton />
+            </ToolbarItem>
+            <ToolbarItem>
+                <Button startIcon={<AddIcon />} component={StackLink} pageName="add" payload="add" variant="contained" color="primary">
+                    <FormattedMessage id="product.newProduct" defaultMessage="New Product" />
+                </Button>
+            </ToolbarItem>
```

Comment:
Buttons to the right of the Toolbar should be wrapped by a single ToolbarActions instead of individual ToolbarItem components.

```suggestion
            <ToolbarActions>
                <GridFilterButton />
                <Button startIcon={<AddIcon />} component={StackLink} pageName="add" payload="add" variant="contained" color="primary">
                    <FormattedMessage id="product.newProduct" defaultMessage="New Product" />
                </Button>
            </ToolbarActions>
```

Reply:
changed